### PR TITLE
Correctly create mocks in TestSuite

### PR DESCRIPTION
**Pull request by antograssiot**
I got all tests passing on PHPUnit 5.5.2 locally, let see what Travis has to say....

### CODE REVIEW

File: tests/TestCase/Database/Schema/MysqlSchemaTest.php [hunk 1]
```diff
@@ -1229,8 +1229,9 @@ public function testDescribeJson()
     protected function _getMockedDriver()
     {
         $driver = new \Cake\Database\Driver\Mysql();
-        $mock = $this->getMockBuilder('FakePdo')
+        $mock = $this->getMockBuilder(\PDO::class)
```
Comment (dereuromark) [inline tests/TestCase/Database/Schema/MysqlSchemaTest.php]: Ideally those two lines would not use FQCN.

Comment (antograssiot) [inline tests/TestCase/Database/Schema/MysqlSchemaTest.php]: @dereuromark I'll update that later and try to improve the use of `::class` where it's possible.
I've no time left today and will be alreday happy if the testsuite is working again 😉

**Comment by antograssiot** [comment]
I'll fix the php 5.5 build tonight

File: tests/TestCase/Database/Schema/MysqlSchemaTest.php (hunk 1 above)
Comment (markstory) [inline tests/TestCase/Database/Schema/MysqlSchemaTest.php]: This is good enough for me.

**Comment by antograssiot** [comment]
updated to please PHP5.5 and PHPUnit 4.8.x
Should be all green now hopefully

**Comment by markstory** [comment]
Awesome work @antograssiot There was one random test failure due to memcached, but I'm not worried about that.

**Pull request merged**